github_issue_summarization/serving_the_model.md

```diff
@@ -1,21 +1,62 @@
 # Serving the model
 
-We are going to use a simple tornado server to serve the model. The [server.py](notebooks/server.py) contains the server code.
+We are going to use [seldon-core](https://github.com/SeldonIO/seldon-core) to serve the model. [IssueSummatization.py](notebooks/IssueSummatization.py) contains the code for this model. We will wrap this class into a seldon-core microservice which we can then deploy as a REST or GRPC API server.
 
-Start the server using `python server.py --port=8888`.
+> The model is written in Keras and when exported as a TensorFlow model seems to be incompatible with TensorFlow Serving. So we're using seldon-core to serve this model since seldon-core allows you to serve any arbitrary model. More details [here](https://github.com/kubeflow/examples/issues/11#issuecomment-371005885).
 
-> The model is written in Keras and when exported as a TensorFlow model seems to be incompatible with TensorFlow Serving. So we're using our own webserver to serve this model. More details [here](https://github.com/kubeflow/examples/issues/11#issuecomment-371005885).
+# Prerequisites
 
-## Sample request
+Ensure that you have the following files from the [training](training_the_model.md) step in your `notebooks` directory:
+
+* `seq2seq_model_tutorial.h5` - the keras model
+* `body_pp.dpkl` - the serialized body preprocessor
+* `title_pp.dpkl` - the serialized title preprocessor
+
+# Wrap the model into a seldon-core microservice
+
+cd into the notebooks directory and run the following docker command. This will create a build/ directory.
+
+```
+cd notebooks/
+docker run -v $(pwd):/my_model seldonio/core-python-wrapper:0.7 /my_model IssueSummarization 0.1 seldonio --base-image=python:3.6
+```
+
+The build/ directory contains all the necessary files to build the seldon-core microservice image
+
+```
+cd build/
+./build_image.sh
+```
+
+Now you should see an image named `seldonio/issuesummarization:0.1` in your docker images. To test the model, you can run it locally using
```

Comment:
Can we tag it into a repo like GCR/DockerHub/Quay? We'll need that to deploy the model on K8s.

Reply:
Updated to use a GCR tag.